src/modules/tests/test-data/cat-be/test-data.cat-be.selector.ts

```diff
@@ -0,0 +1,119 @@
+import { CatBEUniqueTypes } from '@dvsa/mes-test-schema/categories/BE';
+import { CatBELegalRequirements } from '../test-data.models';
+import { QuestionProvider } from '../../../../providers/question/question';
+import { TestCategory } from '../../../../shared/models/test-category';
+import { VehicleChecksQuestion } from '../../../../providers/question/vehicle-checks-question.model';
+
+export const getDrivingFaultCount = (
+    data: CatBEUniqueTypes.TestData, competency: Competencies) => data.drivingFaults[competency];
+
+export const getManoeuvres = (data: CatBEUniqueTypes.TestData): CatBEUniqueTypes.Manoeuvres => data.manoeuvres;
+
+export const hasManoeuvreBeenCompleted = (data: CatBEUniqueTypes.TestData) => {
+  return (
+    get(data.manoeuvres, 'reverseLeft.selected')
+  );
+};
+
+export const hasEyesightTestGotSeriousFault = (data: CatBEUniqueTypes.TestData) => data.eyesightTest.seriousFault;
+
+export const hasLegalRequirementBeenCompleted = (
+  data: CatBEUniqueTypes.TestRequirements, legalRequirement: LegalRequirements) => {
+  return data[legalRequirement];
+};
+
+export const getVehicleChecks = (
+  state: CatBEUniqueTypes.TestData): CatBEUniqueTypes.VehicleChecks => state.vehicleChecks;
+
+export const getTellMeQuestion = (state: CatBEUniqueTypes.VehicleChecks): VehicleChecksQuestion => {
+  const questionProvider: QuestionProvider =  new QuestionProvider();
+  return questionProvider
+    .getTellMeQuestions(TestCategory.BE)
+    .find(question => question.code === get(state, 'tellMeQuestion.code'));
+};
+
+export const areTellMeQuestionsSelected = (
+  state: CatBEUniqueTypes.VehicleChecks) => typeof get(state, 'tellMeQuestions') !== 'undefined';
+
+export const areTellMeQuestionsCorrect = (state: CatBEUniqueTypes.VehicleChecks) => {
+  const tellMeQuestions = get(state, 'tellMeQuestions');
+  let correct = true;
+
+  if (typeof tellMeQuestions === 'undefined' || tellMeQuestions === null || !(tellMeQuestions instanceof Array)) {
+    correct = false;
+  } else {
+    tellMeQuestions.forEach((question) => {
+      if (question.outcome !== CompetencyOutcome.P) {
+        correct = false;
+      }
+    });
+  }
+
+  return correct;
+};
+
+export const getCatBELegalRequirements = (data: CatBEUniqueTypes.TestData): CatBELegalRequirements => {
+  return {
+    normalStart1: data.testRequirements.normalStart1 || false,
+    normalStart2: data.testRequirements.normalStart2 || false,
+    downhillStart: data.testRequirements.downhillStart || false,
+    uphillStart: data.testRequirements.uphillStart || false,
+    angledStartControlledStop: data.testRequirements.angledStartControlledStop || false,
+    manoeuvre: hasManoeuvreBeenCompleted(data) || false,
+    vehicleChecks: hasVehicleChecksBeenCompleted(data) || false,
+    eco: data.eco.completed || false,
+  };
+};
+
+export const hasVehicleChecksBeenCompleted = (data: CatBEUniqueTypes.TestData): boolean => {
+  const REQUIRED_SHOW_ME_QUESTIONS = 3;
```

Comment:
Think similar constants have been created as part of the showme tellme changes. (I know it wasn't available when you were writing this)

Reply:
Done